crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -330,6 +361,76 @@ impl<S: DatabaseExt + Send, H: HistoryMode> DynTracer<EraDb<S>, SimpleMemory<H>>
+                        .callstack
+                        .inner
+                        .last()
+                        .expect("callstack before the current");
+
+                    if state.vm_local_state.callstack.depth() == broadcast.depth // + 1
+                        &&
+                        prev_cs.this_address == broadcast.original_caller
+                    {
+                        self.one_time_actions.push(FinishCycleOneTimeActions::SetOrigin {
+                            origin: broadcast.new_origin,
+                        });
+
+                        let new_origin = broadcast.new_origin;
+                        let revm_db_for_era = storage
+                            .borrow_mut()
+                            .storage_handle
+                            .inner
+                            .read()
+                            .unwrap()
+                            .fork
+                            .as_ref()
+                            .unwrap()
+                            .fork_source
+                            .clone();
+                        let rpc = revm_db_for_era.db.lock().unwrap().active_fork_url();
+
+                        let farcall_abi = {
+                            let src0_idx = data.opcode.src0_reg_idx as usize;
+                            let src0 = state.vm_local_state.registers[src0_idx];
+
+                            FarCallABI::from_u256(src0.value)
+                        };
+
+                        let gas_limit = current.ergs_remaining;
+                        let (nonce, _) = Self::get_nonce(new_origin, &mut storage.borrow_mut());
+
+                        let tx = BroadcastableTransaction {
+                            rpc,
+                            transaction:
+                                ethers::types::transaction::eip2718::TypedTransaction::Legacy(
+                                    TransactionRequest {
+                                        from: Some(new_origin),
+                                        to: Some(ethers::types::NameOrAddress::Address(
+                                            current.code_address,
+                                        )),
+                                        //FIXME: set only if set manually by user
+                                        gas: Some(gas_limit.into()),
+                                        //FIXME: retrieve proper value
+                                        value: Some(farcall_abi.ergs_passed.into()),
+                                        data: Some(get_calldata(&state, memory).into()),
+                                        nonce: Some(nonce),
+                                        ..Default::default()
+                                    },
+                                ),
+                        };
+                        tracing::debug!(?tx, "storing for broadcast");
+
+                        self.broadcastable_transactions.push(tx);
```

Comment:
Are we using this variable somehow, or is it for a later PR?

Reply:
`tx` no, it's just for debugging, meanwhile `broadcastable_transactions` is the actual output of the cheatcode

Reply:
Wondering how these transactions would be applied

Reply:
This information is used in scripting